cli_tools/osconfig_agent/tasker/tasker.go

```diff
@@ -0,0 +1,48 @@
+//  Copyright 2018 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+// Package tasker is a task queue for the osconfig_agent.
+package tasker
+
+import "github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/osconfig_agent/logger"
+
+var tc chan *task
+
+func init() {
+	tc = make(chan *task)
+	go tasker()
+}
+
+type task struct {
+	name string
+	run  func()
+}
+
+// Enqueue adds a task to the task queue.
+func Enqueue(name string, f func()) {
+	tc <- &task{name: name, run: f}
+}
+
+func tasker() {
+	logger.Debugf("Tasker start.")
+	for {
+		logger.Debugf("Waiting for tasks to run.")
+		select {
+		case t := <-tc:
+			logger.Debugf("Tasker running %q.", t.name)
```

Comment:
Do log messages have a timestamp? It may be useful to know how long tasks take to execute.

Reply:
Yes, right now all logs use these flags:
log.Ldate | log.Lmicroseconds | log.Lshortfile

https://golang.org/pkg/log/#pkg-constants